test/results_test.exs

```diff
@@ -58,13 +68,118 @@ defmodule Wanda.ResultsTest do
                %ExecutionResult{
                  execution_id: ^execution_id,
                  group_id: ^group_id,
-                 payload: %{
-                   "result" => "passing",
-                   "check_results" => [_ | _]
-                 }
+                 status: :running,
+                 payload: %{},
+                 targets: [
+                   %{
+                     agent_id: ^agent_id_1,
+                     checks: ^checks_1
+                   },
+                   %{
+                     agent_id: ^agent_id_2,
+                     checks: ^checks_2
+                   }
+                 ]
                }
              ] = Repo.all(ExecutionResult)
     end
+
+    test "should raise an error when trying to create an already existing execution" do
+      [
+        %ExecutionResult{execution_id: execution_id, group_id: group_id},
+        %ExecutionResult{}
+      ] = insert_list(2, :execution_result)
+
+      assert_raise Ecto.ConstraintError, fn ->
+        Results.create_execution_result!(execution_id, group_id, build_list(2, :target))
+      end
+
+      assert 2 =
+               ExecutionResult
+               |> Repo.all()
+               |> length()
+    end
+  end
+
+  describe "Completing an Execution" do
+    test "should correctly complete a running execution" do
+      %ExecutionResult{
+        execution_id: execution_id,
+        group_id: group_id,
+        status: :running,
+        payload: %{}
+      } = insert(:running_execution_result)
+
+      assert %ExecutionResult{
+               execution_id: ^execution_id,
+               group_id: ^group_id,
+               status: :completed,
+               payload: %{
+                 result: :passing
+               }
+             } =
+               Results.complete_execution_result!(
+                 build(
+                   :result,
+                   execution_id: execution_id,
+                   group_id: group_id,
+                   result: :passing
+                 )
+               )
+    end
+
+    test "should pass through an already completed execution" do
```

Comment:
What is this test for? To check if already completed executions are not changed?
It looks way to complicated, doesn't it?
If we want to check that the payload is changed, I would do it in the previous test, which is the happy path

Reply:
+ 1 on removing this test, we already know that if we update a row in the database with the same data, the operation is idempotent.

Reply:
Adjusted here https://github.com/trento-project/wanda/pull/60/commits/a272e9ce8df9bd3dcb4de5e0ee618ffa3e67472e